src/box/memtx_engine.cc

```diff
@@ -1227,11 +1386,16 @@ memtx_engine_wait_checkpoint(struct engine *engine,
 static ssize_t
 memtx_engine_commit_checkpoint_f(va_list ap)
 {
-	struct xlog *xlog = va_arg(ap, typeof(xlog));
-	if (xlog_materialize(xlog) != 0) {
+	struct checkpoint *ckpt = va_arg(ap, typeof(ckpt));
```

Comment:
Nit: IMHO, it would be better to do this refactoring (passing `checkpoint` instead of `xlog` to these methods) in a separate commit.

Reply:
This does not seem right to me TBH: it's a 2-line change required to have the specific data available in the function (4 lines along with the same suggestion below).

Reply:
Done.